### PR TITLE
chore(storage): reduce reschedule when cancel task

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

as title

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- avoid reschedule pick compaction when NoAvailCompactor

## Refer to a related PR or issue link (optional)
- https://github.com/risingwavelabs/risingwave/issues/5303

### CODE REVIEW

**Review by zwang28 — APPROVED** [review]

File: src/meta/src/hummock/manager/mod.rs [hunk 1]
```diff
@@ -1045,7 +1045,15 @@ where
             compact_task.compaction_group_id,
         );
 
-        self.try_send_compaction_request(compact_task.compaction_group_id)?;
+        if !CACEL_STATUS_SET.contains(&compact_task.task_status()) {
```
Comment (hzxa21) [inline src/meta/src/hummock/manager/mod.rs:1048]: nits: should be `CANCEL_STATUS_SET`

**Comment by Li0k** [comment]
https://github.com/risingwavelabs/risingwave/pull/5405

File: src/meta/src/hummock/manager/mod.rs [hunk 2]
```diff
@@ -1045,7 +1045,15 @@ where
...
+            // We no need to reschedule when report cancel_task
+            // In fact, when `cancel_task` appears in the system, it means that the system
+            // processing capacity may have been overloaded, and we may no longer need to try to
+            // schedule. And this is a quick fix
+            // TODO: need to consider the fairness of the compaction_group and unload the schedule
+            // when overloaded
+            self.try_send_compaction_request(compact_task.compaction_group_id)?;
```
Comment (hzxa21) [inline src/meta/src/hummock/manager/mod.rs:1055]: I think it still makes sense to retry compaction when we have task cancelled in some cases:
- SendFailCanceled: this happens when the rpc between meta and a picked compactor fails but if there are other available compactors, we can still trigger compaction so skip this case brings no benefit.
- Same for HeartbeatCanceled

Given that the issue caused by rescheduling tasks when compactor is not available is partially fixed by #5405. I think we can skip this PR.

Comment (Gun9niR) [reply src/meta/src/hummock/manager/mod.rs:1055]: I think with schedule state purposed in #5321, we can send the request regardless.

**Pull request closed without merging**